src/main/java/dao/WalletDAO.java

```diff
@@ -0,0 +1,157 @@
+package dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class WalletDAO {
+    //    this class contains methods to process wallet and puchase artifacts
+
+    //TODO Get DBCreator object to private filed of WallDao class instead of creating it in every method
+    DBCreator dbCreator;
+
+    WalletDAO(){
+        dbCreator = new DBCreator();
+    }
+
+    public int showWallet(int id) throws SQLException {
```

Comment:
Could be replaced with universal method getCodecooler() in CodecoolerDAO, so we can use getter on the returned object Codecooler. Same goes to method seeStudentsWallet, which returns map.

Reply:
hey, interesting idea about using getters instead of method showWallet, but at this point I just want to focus on structure of new DAO, so we can merge it into master and continue to work. Do you think that now it is clear and consistent. I don't want to remove any methods - just change their places.  

Your approach to showWallet/s method seems fine - I think that when we were writing all those methods we just focus on writing sql queries and forget about java layer. But as for as now I think we shouldn't remove anything (who knows what can be useful later). Next week when we will connect dao with contorollers we should try to do this as You wrote.

But I'am not sure about that:
"WalletDAO isn't really needed, as should have every element we need in Codecooler object. All operations on coins should be in controller or some kind of service."

maybe it would be possible to create all buying sequence only with getters and setters to codecoller and artifact but when I think about that I see problem: in this approach to buy artifact you pass to some method (buyAtifact()) artifact object and codecooler object which have to be created at some moment before calling buying method - what if in time between creating and buying sth has changed in database? We can ask  mentors next week about it.
Still we need to write some update methods for tables and we need to store then somewhere